test/models/drive_test.dart

```diff
@@ -0,0 +1,1003 @@
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.pending,
+            seats: 3),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByRider,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.rejected,
+            seats: 2),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.withdrawnByRider,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.cancelledByDriver,
+            seats: 2),
+      ]);
+      expect(drive.getMaxUsedSeats(), 1);
+    });
+    test('returns 2 if rides has 2 approved ride with 1 seat at the same time', () async {
+      final Drive drive = DriveFactory().generateFake(seats: 3, rides: <Ride>[
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14),
+            endTime: DateTime(2022, 2, 2, 14, 20),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 15),
+            endTime: DateTime(2022, 2, 2, 15, 30),
+            status: RideStatus.approved,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.withdrawnByRider,
+            seats: 1),
+        RideFactory().generateFake(
+            startTime: DateTime(2022, 2, 2, 14, 30),
+            endTime: DateTime(2022, 2, 2, 15),
+            status: RideStatus.pending,
+            seats: 1),
+      ]);
+      expect(drive.getMaxUsedSeats(), 2);
+    });
+    test('returns 4 if rides has a complex overlay where at its peak has 4 seats taken by approved rides', () async {
```

Comment:
I think in this case the point can be made that it's fair to test a more complex case, but please at least make a comment which rides overlap here so that I don't have to read all of this code (DateTimes aren't that easy to read)

Reply:
Also, and I think this goes for all of these tests, the description isn't really ideal, IMO. I would think about what I'd want to see in a console if a test fails to know exactly where it has failed, quickly. In this case, I'm not interested how many seats there are and what is expected; this information is already provided in the code once I go look for the test. What I'd want to know is that the complex Drive.getMaxUsedSeats test has failed. In the test above, I'd want to know that the overlapping getMaxUsedSeats test has failed, and in the test above that I'd want to know that the non-overlapping getMaxUsedSeats test has failed. I think naming tests like this might already help determining useful test cases